http/http-advanced/src/main/java/io/quarkus/ts/http/advanced/GrpcClientInterceptors.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.ts.http.advanced;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.MethodDescriptor;
+import io.quarkus.grpc.GlobalInterceptor;
+
+class GrpcClientInterceptors {
+    @GlobalInterceptor
+    @ApplicationScoped
+    static class TypeTarget extends Base {
+    }
+
+    static class MethodTarget extends Base {
+    }
+
+    static class Producer {
+        @GlobalInterceptor
+        @Produces
+        MethodTarget methodTarget() {
+            return new MethodTarget();
+        }
+    }
+
+    abstract static class Base implements ClientInterceptor {
+        @Override
+        public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method, CallOptions options,
+                Channel next) {
+            GrpcResource.invoked.add(getClass().getName());
```

Comment:
add new headers, one example I found in a second was `io.quarkus.grpc.examples.hello.HeaderCollectingInterceptor` looks like it comply your use case, but you will know better